auth_app/repositories/users.py

```diff
@@ -1,37 +1,82 @@
-from sqlalchemy import delete, insert, select, update
-from sqlalchemy.ext.asyncio import AsyncSession
+from typing import Optional
+from uuid import UUID
+
+from sqlalchemy import and_, delete, insert, select, update
 
 from auth_app.models.users import users
-from auth_app.schemas.sign_up import (
+from auth_app.repositories.base import BaseRepo
+from auth_app.schemes.users import (
     CreateUser,
-    DeleteUser,
     GetUser,
-    PatchUser,
-    PutUser,
+)
+from auth_app.services.utils.pwd_hashing import (
+    hash_password,
 )
 
 
-class UserSignUpRepo:
-    def __init__(self, session: AsyncSession):
-        self.session = session
+class UserRepo(BaseRepo):
 
-    async def create(self, create_data: CreateUser) -> GetUser:
-        stmt = insert(users).values(
-            **create_data.model_dump()
-        ).returning(*users.c)
+    async def create_user(self, create_data: CreateUser) -> GetUser:
+        data = create_data.model_dump()
+        data['password_hash'] = hash_password(data.get('password_hash'))
+        stmt = insert(users).values(**data).returning(*users.c)
         result = await self.session.execute(stmt)
         await self.session.commit()
         row = result.mappings().one()
         return GetUser.model_validate(row)
 
-    async def get(self, user_id: int) -> GetUser | None:
-        stmt = select(users).where(users.c.id == user_id)
+    async def get_user(self, email: str) -> Optional[GetUser]:
+        stmt = select(users).where(users.c.email == email)
         result = await self.session.execute(stmt)
-        row = result.mappings().one()
-        return GetUser.model_validate(row)
+        row = result.mappings().first()
+        return GetUser.model_validate(row) if row else None
+
+    async def get_users(self, filter_dict: Optional[dict]) -> list[GetUser]:
+        stmt = select(users)
+        conditions = []
+        if not filter_dict:
+            filter_dict = {}
+        for k, v in filter_dict.items():
+            column = getattr(users.c, k, None)
+            if column is not None:
+                conditions.append(column == v)
+        if conditions:
+            stmt = stmt.where(and_(*conditions))
+        result = await self.session.execute(stmt)
+        rows = result.mappings().fetchall()
+        return [GetUser.model_validate(prt) for prt in rows]
+
+    async def put_user(
+        self, user_id: UUID, put_dict: dict
+    ) -> Optional[GetUser]:
```

Comment:
Начиная в питон 3.10 лучше писать аннотации иначе
```suggestion
    ) -> GetUser | None:
```

Reply:
согласен, ранее только в пайдентике юзал, потом почему-то начал и в других местах, поправил, в пайдентик схемах оставил, мне там чуть больше нравится данный подход, но если критично - уберу и там)